pandas/tests/frame/indexing/test_where.py

```diff
@@ -328,15 +328,14 @@ def test_where_bug_mixed(self, any_signed_int_numpy_dtype):
         )
 
         expected = DataFrame(
-            {"a": [np.nan, np.nan, 3.0, 4.0], "b": [4.0, 3.0, np.nan, np.nan]},
```

Comment:
changing this test to fill with -1 instead of with nan

Reply:
just checking: you looked at the original issue and confirmed this still tests the relevant bug?

Reply:
yeah the original issue is just about having mixed dtypes https://github.com/pandas-dev/pandas/issues/2793 , I've preserved that but have just updated the fill value to one which won't upcast